designs/sam-debugging/local-sam-debugging.md

```diff
@@ -0,0 +1,308 @@
+
+When clicked, the Run and Debug CodeLenses locally invoke their associated Template Resource. The following takes place:
+
+-   The SAM Application is built
+-   The associated SAM Template resource is invoked, using configurations set with the Configure CodeLens
+-   (If the Debug CodeLens was clicked) The VS Code debugger is attached to the invoked resource
+
+When clicked, the Configure CodeLens opens a (JSON) configuration file that resides in the workspace and is read by the Toolkit. The configuration file is used for each SAM Template Resource within the workspace. Users have autocompletion support in this file. A rich UI for configuration is not considered at this time, but the door remains open to adding a visual editor in the future based on user feedback.
+
+The Run and Debug CodeLenses perform a regular local invoke on a SAM Template resource. These CodeLenses do not perform API Gateway style invokes.
+
+#### CodeLenses in Code files
+
+CodeLenses in code files provide support for debugging Standalone Lambda function handlers.
+
+The following CodeLenses appear over any function that appears to be an eligible Lambda Handler:
+
+-   Run Locally - See below for details
+-   Debug Locally - See below for details
+-   Configure - allows the user to configure arguments that are used with the Run and Debug CodeLenses (see [What can be configured for a Debug session?](#debug-config))
+
+When clicked, the Run and Debug CodeLenses locally invoke the Lambda handler function they represent. These Lambda handlers are invoked independent of SAM Templates that exist in the users workspace. The following takes place:
+
+-   A temporary SAM Template is produced, which contains one resource that references the Lambda handler
+-   The temporary SAM Application is built
+-   The resource in the temporary SAM Template is invoked, using configurations set with the Configure CodeLens
+-   (If the Debug CodeLens was clicked) The VS Code debugger is attached to the invoked resource
+
+When clicked, the Configure CodeLens opens a (JSON) configuration file that resides in the workspace and is managed by the Toolkit. All standalone handlers within a workspace will have their configurations stored in this file. Users have autocompletion support in this file. A rich UI for configuration is not considered at this time, but the door remains open to adding a visual editor in the future based on user feedback.
+
+These CodeLenses do not support API Gateway style invokes.
+
+Some users may find CodeLenses within code files distracting, particularly if they are using the Toolkit for features not related to local debugging. Toolkit settings can be used to enable and disable CodeLenses. CodeLenses only appear for languages/runtimes that the Toolkit has implemented Debug support for.
+
+### Graphical User Interface
+
+SAM Template Resources invoked by API Gateway can be locally tested with a REST Workbench.
```

Comment:
Is "REST Workbench" a VSCode-ism?

Reply:
No. It's intended to represent this UI feature, something akin to Postman. Revision suggestions?